### ISSUE TITLE
jsdoc/require-jsdoc do not recgonize the export typr 

### ISSUE THREAD

**Issue by IsaacIsrael** [gajus/eslint-plugin-jsdoc#1319]
## Expected behavior

```typescript

type Props = {
  readyonly variant:  string
}

export type { Props as ComponentProps };
```

## Actual behavior

```typescript

// Do not recognize as error to pe enter the documentation
type Props = {
  readyonly variant:  string
}

export type { Props as ComponentProps };
```

## ESLint Config

```js 
{
"jsdoc/require-throws": "error",
      "jsdoc/check-indentation": "warn",
      "jsdoc/no-blank-blocks": "warn",
      "jsdoc/require-asterisk-prefix": "warn",
      "jsdoc/require-description": "warn",
      "jsdoc/sort-tags": "warn",
      "jsdoc/check-syntax": "warn",
      "jsdoc/tag-lines": ["warn", "never", { startLines: 1 }],
      "jsdoc/require-param": ["warn", { checkDestructuredRoots: false }],
      "jsdoc/require-jsdoc": [
        "warn",
        {
          publicOnly: { ancestorsOnly: false},
          require: {
            FunctionDeclaration: true,
            FunctionExpression: true,
            ArrowFunctionExpression: true,
            ClassDeclaration: true,
            ClassExpression: true,
            MethodDefinition: true,
          },
          contexts: [
            "VariableDeclaration",
            "TSTypeAliasDeclaration",
            // Encourage documenting React prop types
            "TSPropertySignature",
            "TSInterfaceDeclaration",
            "TSMethodSignature",
            "TSEnumDeclaration"
          ],
          enableFixer: true,
        },
      ],
      // tsdoc checks this syntax instead
      "jsdoc/require-hyphen-before-param-description": "off",
      "jsdoc/require-returns": "off",
}
```

## Environment

- Node version: v20.17.0
- ESLint version  v8.57.1
- `eslint-plugin-jsdoc` version: ^50.2.4

**Comment by brettz9** [comment]
You have `ancestorsOnly` set, so it wouldn't find it anyways (that is an optimization to only find if the node is directly nested inside an export). Setting it instead to `esm: true` should in theory work, but it indeed does look like that is a bug. #1320 will hopefully address.

**Comment by IsaacIsrael** [comment]
he @brettz9  thank for the feedback but this did not worked for me :(

**Comment by brettz9** [comment]
There was a bug and I provided a fix, but the release did not successfully go through. @gajus ?

**Comment by gajus** [comment]
Looks like it was broken in this commit/PR

https://github.com/gajus/eslint-plugin-jsdoc/commit/239d69aafff5462852a4c73c2ea4ab0a2e84a35c

Reverted and it works
